design/user.go

```diff
@@ -0,0 +1,145 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var _ = a.Resource("user", func() {
+	a.BasePath("/user")
+
+	a.Action("show", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET(""),
+		)
+		a.Description("Get the authenticated user in JSON-API format")
+		a.UseTrait("conditional")
+		a.Response(d.OK, showUser)
+		a.Response(d.NotModified)
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+	})
+
+	a.Action("listResources", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/resources"),
+		)
+		a.Params(func() {
+			// requirement and value will be handled by the controller method,
+			// in order to be able to return a proper JSON-API response to the client in case
+			// of a bad request
+			a.Param("type", d.String, "the type of resource to list")
+			a.Required("type")
+		})
+		a.Description("List resources of a given type with a role for the current user")
+		// a.UseTrait("conditional")
```

Comment:
sure, that was a left-over since support for conditional requests will be implemented in https://github.com/fabric8-services/fabric8-auth/issues/606 - just opened, forgot about it before so thanks for reminding me about it! ;)